### ISSUE TITLE
Server config was running fine, then all of a sudden I get this error. Uninstall and reinstall fresh, and still get errors on clean install. [CARROT-982]

### ISSUE THREAD

**Issue by dweiss** [carrot2/carrot2#1179]
I had the search engine running like I wanted. I returned to the office the next week, and started seeing these errors. 

Attribute binding failed: No value for required attribute: OpenSearchDocumentSource.feedUrlTemplate (org.carrot2.source.opensearch.OpenSearchDocumentSource#feedUrlTemplate. 

I checked and rechecked the FeedURL changing it multiple times, but I still get the same error. 

I went ahead and uninstalled everything including Tomcat. I went ahead and reinstalled, and when doing a search with the Defualt out of the box configuration, I get the following error.  

Attribute binding failed: No value for required attribute: Bing3DocumentSource.appid (org.carrot2.source.microsoft.Bing3DocumentSource#appid)

Is something messed up with Java? Have you seen this before. What could be missing?

I've attached a file with the opensearch attributes I am using.

---
Issue: CARROT-982 (migrated from JIRA), created by Ricardo Vallecillo, resolved Jan 24 2013
Attachments: [source-opensearch-attributes.xml](https://carrotsearch.github.io/jira-archive-public/attachments/carrot2/CARROT-982/source-opensearch-attributes.xml)

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

This error:

```Java
Attribute binding failed: No value for required attribute: Bing3DocumentSource.appid (org.carrot2.source.microsoft.Bing3DocumentSource#appid)
```


is caused by the fact that Bing document source now requires that you pass a valid application ID to it (if you're using it then head on to http://download.carrot2.org/stable/javadoc/org/carrot2/source/microsoft/Bing3DocumentSource.html#appid follow the link and generate your appid, then pass it via a system property mentioned in the JavaDoc.

As for the first error – I have no clue, looks good to me. Let me check.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Here's what I did:

1. fetched the webapp from: http://project.carrot2.org/download-webapp.html
2. used tomcat 7.0.32 that I had installed locally
3. moved the webapp under tomcat/webapps
4. opened http://localhost:8080/carrot2-webapp-3.6.2/search clicked on the 'web' tab, ran a query. worked. 
5. the bing tab didn't work for the reason above (no API key, fine).
6. I then stopped the webapp, removed the webapps/\*.war file and replaced the source-indeed-attributes.xml with your attributes XML in the unpacked context for simplicity.
7. started tomcat again and accessed the 'indeed' tab which should not point at your opensearch server. I got an error:

```Java
2013-01-22 21:13:54,986,[ERROR],[http-bio-8080-exec-3],org.carrot2.webapp.QueryProcessorServlet,Processing error: searchdev10.na.atkinsglobal.com
org.carrot2.core.ProcessingException: searchdev10.na.atkinsglobal.com
...
Caused by: java.net.UnknownHostException: searchdev10.na.atkinsglobal.com
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:157)
```

but this seems to be caused by the inability to resolve the search host, not the infrastructure problem.
